protocol/tx/map.go

```diff
@@ -33,8 +33,13 @@ func mapTx(tx *bc.TxData) (headerID bc.Hash, hdr *header, entryMap map[bc.Hash]e
 
 	for i, inp := range tx.Inputs {
 		if oldSp, ok := inp.TypedInput.(*bc.SpendInput); ok {
+			var oldSpID bc.Hash
+			oldSpID, err = ComputeOutputID(&oldSp.SpendCommitment)
```

Comment:
In the absence of distinguishing types for outputIDs and IDs of other kinds of entry, please observe a convention of putting "out" somewhere in the varname when it holds an outputID. As it is, `oldSpID` looks like it could be the ID of a spend entry.